### ISSUE TITLE
[RFC] The case for special-case IDs

### ISSUE THREAD

**Issue by stairman06** [modip-org/spec#12]
Widespread adoption of the MODIP format is key to success. Getting hosts to serve MODIP metadata is, in my opinion, a crucial part that we need to do right. The format needs to be easy to implement from a server-side perspective, otherwise we gain no adoption. A format that's hard, tricky, or cumbersome to implement doesn't gain adoption.

## Framework Overhead
Right now, the format doesn't have a ton of overhead. However, one of the current issues is with regards to modloaders/frameworks. The current specification requires frameworks to be stored just like a regular project would - and this is great from a technical perspective. It grants lots of freedom, leeway, and extensibility. But we have more to think about than just technical issues. Somebody has to be providing the metadata for the modloaders.

The requirement to serve metadata for modloaders is, frankly, a burden. Hosts don't have any incentive to - it's not their job, and convincing the developers of modloaders to serve MODIP metadata is basically impossible. And this is why I think it may be okay to grant a hopefully-temporary special exemption to some frameworks.

## The Exemption
The idea is to grant special exemptions for certain IDs. This may sound like an incredibly stupid idea at first, but bear with me. I believe these exemptions could be crucial to adoption. This would prevent dependencies with the ID of `framework-forge` and `framework-fabric-loader` from needing a source project specified. Nobody has to host it, a launcher can fetch it themselves using their own methods.

Exemptions are nothing new, and would probably have been done anyway, although unofficially. If a launcher already knows of Fabric and where to get it, it has no reason to fetch potentially-outdated metadata from somewhere else. The same thing goes for Forge, which is an even larger issue as it uses a complicated installation process that can't be represented in MODIP metadata. Launchers would likely special case the Forge ID anyway.

## Other Solutions
Other possible solutions that aren't exemptions:
- Have us (MODIP as a group) provide the metadata. We would do the work to convert existing modloaders into MODIP-compliant metadata. Hosts can then link to a URL where we have the metadata served.
-  Encourage hosts to do it themselves. This, of course, is a harder solution. We could offer to help write tools that could handle automatically fetching non-MODIP metadata, converting it, and serving it from the hosts URL.

I welcome all feedback and discussion. This is a complicated issue.

**Comment by expeehaa** [comment]
Special cases almost always make things more difficult and have their drawbacks, most notably that they require a different implementation. I think the spec should not default to special cases to minimize the requirements for having a complete implementation.
Instead, if an implementator wants their implementation to treat certain IDs specially, they may implement it that way. But that doesn't force anyone to do the same.

**Comment by immibis** [comment]
Everything that you've said about frameworks also applies to non-framework mods. Should there be a special ID for TechReborn?

**Comment by stairman06** [comment]
Yeah, but it's most common for frameworks. Nearly everything will require a framework of some sort. That being said, maybe there should be a special type for these standardized dependencies (e.g. `"type": "wellKnown"`)
